docs/design/Network Topology Aware Scheduling.md

```diff
@@ -0,0 +1,575 @@
+![nework topology generation](images/network-topology-aware/nework-topology-generation.png)    
+
+### Job management
+
+HyperJob is an API for managing a group of replicated volcano jobs, which aligns with HyperNode/SuperPod for optimal LLM training.
+
+```go
+type HyperJob struct {  
+	metav1.TypeMeta `json:",inline"\`
+
+	// +optional  
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"\`
+
+	// Specification of the desired behavior of the volcano Hyperjob, including the minAvailable  
+	// +optional  
+	Spec HyperJobSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"\`
+
+	// Current status of the volcano HyperJob  
+	// +optional  
+	Status HyperJobStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"\`  
+}
+
+type HyperJobSpec struct {  
+	// The minimal available Job to run for this HyperJob  
+	// Defaults to the summary of jobs' replicas  
+	// +optional  
+	MinAvailable int32 `json:"minAvailable,omitempty" protobuf:"bytes,2,opt,name=minAvailable"\`  
+	  
+	Jobs []JobSepc `json:"jobs,omitempty" protobuf:"bytes,4,opt,name=jobs"\`  
+	  
+...  
+}
+```
+
+##### Volcano Job sample:
+
+```go
+apiVersion: batch.volcano.sh/v1alpha1  
+kind: Job  
+metadata:  
+  name: pytorch-job  
+spec:  
+  minAvailable: 1  
+  schedulerName: volcano  
+  // describes the network topology scheduling requirement of jobs  
+  // example: hard, with highest-tier-allowed: 1, means for each job-foo, scheduler needs to find a tier1 to run all the job-foo pods.   
+  networkTopologies:  
+  - mode: hard 
+    highestTierAllowed: 1
+  plugins:  
+    pytorch: ["--master=master","--worker=worker","--port=23456"] # Pytorch plugin register  
+  tasks:  
+    - replicas: 1  
+      name: master  
+      policies:  
+        - event: TaskCompleted  
+          action: CompleteJob  
+      template:  
+        spec:  
+          containers:  
+            - image: gcr.io/kubeflow-ci/pytorch-dist-sendrecv-test:1.0  
+              imagePullPolicy: IfNotPresent  
+              name: master  
+          restartPolicy: OnFailure  
+    - replicas: 2  
+      name: worker  
+      template:  
+        spec:  
+          containers:  
+            - image: gcr.io/kubeflow-ci/pytorch-dist-sendrecv-test:1.0  
+              imagePullPolicy: IfNotPresent  
+              name: worker  
+              workingDir: /home  
+          restartPolicy: OnFailure
+```
+
+##### HyperJob sample:
+
+```go
+apiVersion: [batch.volcano.sh/v1aplha1](http://batch.volcano.sh/v1aplha1)  
+kind: HyperJob  
+metadata:  
+    name: multi-volcano-job  
+spec:  
+    networkTopologies: 
+    - mode: hard** // we don’t really need to explicitly indicate the soft requirements, we can make it as a default algorithm behavior. And we need to describe the default behavior in the field comment  
+      highestTierAllowed: 2**  
+    replicatedJobs:  
```

Comment:
The fields of this HyperJob sample did not match the API design.

Reply:
done